packages/wallet/src/redux/__tests__/helpers.ts

```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
+        expect.objectContaining({
+          messagePayload: expect.objectContaining({
+            signedCommitment: transformCommitmentToMatcher({ commitment }),
+          }),
+        }),
+      ]),
     );
-  });
-};
+  } catch (err) {
+    if ('matcherResult' in err) {
+      // In this case, we've caught a jest expectation error.
+      // We try to help the developer by using expect(foo).toMatchObject(bar)
+      // The errors are much more useful in this case, but will be deceiving in the case when
+      // multiple messages are queued.
+
+      // To help with debugging, you can change the idx variable when running tests to 'search'
+      // for the correct commitment
 
-export const expectThisCommitmentSent = (state: SideEffectState, c: Partial<Commitment>) => {
-  expectThisMessageAndCommitmentSent(state, c, 'WALLET.COMMON.COMMITMENT_RECEIVED');
+      expect(messageOutbox[idx]).toMatchObject({
+        messagePayload: {
+          type,
+          signedCommitment: { commitment },
+        },
+      });
+    } else {
+      throw err;
+    }
+  }
 };
-type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
 
 export const expectTheseCommitmentsSent = (
   state: SideEffectState,
   commitments: PartialCommitments,
+  type = 'WALLET.COMMON.COMMITMENTS_RECEIVED',
+  idx = 0,
 ) => {
-  expectThisMessageAndTheseCommitmentsSent(
-    state,
-    commitments,
-    'WALLET.COMMON.COMMITMENTS_RECEIVED',
-  );
+  const messageOutbox = getOutboxState(state, 'messageOutbox');
+
+  try {
+    // Passes when at least one message matches
+    // In the case of multiple messages queued, this approach does not care about
+    // their order, which is beneficial.
+    // However, the diffs produced by jest are unreadable, so when this expectation fails,
+    // we catch the error below
+    expect(messageOutbox).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          messagePayload: expect.objectContaining({
+            signedCommitments: commitments.map(transformCommitmentToMatcher),
+          }),
+        }),
+      ]),
+    );
+  } catch (err) {
+    if ('matcherResult' in err) {
+      // In this case, we've caught a jest expectation error.
+      // We try to help the developer by using expect(foo).toMatchObject(bar)
+      // The errors are much more useful in this case, but will be deceiving in the case when
+      // multiple messages are queued.
+
+      // To help with debugging, you can change the idx variable when running tests to 'search'
+      // for the correct commitment
+      expect(messageOutbox[idx]).toMatchObject({
+        messagePayload: {
+          type,
+          signedCommitments: commitments,
+        },
+      });
+    } else {
+      throw err;
+    }
+  }
 };
 
+function getOutboxState(state: SideEffectState, outboxBranch: 'messageOutbox'): MessageOutbox {
+  if ('sideEffects' in state && state.sideEffects && state.sideEffects[outboxBranch]) {
```

Comment:
Why is `'sideEffects' in state` needed?

Reply:
`SideEffectState` is a union type, where not all members have the `sideEffect` property. So, without, the compiler is unaware that `state` can have a `sideEffect` property.